homebridge-ui/public/index.html

```diff
@@ -211,36 +241,47 @@ <h5 class="mb-0 text-center">
             addCell.appendChild(document.createTextNode('Not supported'));
           } else if (!validAuth) {
             addCell.appendChild(document.createTextNode('Invalid credentials'));
-          } else if (currentDevice = currentConfig.devices.find((d) => d.id === device.id)) {
-            // addCell.appendChild(document.createTextNode('Already added'));
-            debugLog(`Existing device:\n${JSON.stringify(currentDevice, null, 2)}`);
+          } else if (currentConfig.devices.find((d) => d.id === device.id)) {
+            // We already have a record for this device, either with device ID, add button to update it.
             const button = document.createElement('button');
             button.innerText = 'Update';
             button.className = 'btn btn-outline-secondary btn-sm';
             button.addEventListener('click', async () => {
-              const i = configuration.devices.findIndex((o) => o.id === device.id);
-              configuration.devices[i].advanced_options.token = device.token;
-              configuration.devices[i].advanced_options.key = device.key;
-              createForm(configSchema, configuration); // update view
-
-              addCell.removeChild(button);
-              addCell.appendChild(document.createTextNode('Updated'));
-
-              homebridge.toast.success('Device Updated');
+              // Update button clicked...
+              const viewDev = configuration.devices.find((d) => d.id === device.id);
```

Comment:
Why create a new variable? I think if you copy a device from the list it won't affect the list itself (so it's not a reference). I might be wrong, but I prefer the method I made.

Reply:
I have put back to how it was, but kept a check that the device was found in the array.  Not sure if I'm being over cautious or not.

Reply:
I think you are. It's ensured that the device exists because of this line https://github.com/kovapatrik/homebridge-midea-platform/pull/38/commits/9b021a09c78f1a74cfb76f9ade162ce489079d9f#diff-e555da33a837dc0c5eb245245418f0de2e0b78f22bc57980bdd626f7548125fcR244

Reply:
yes you're right.  Will remove the test.